lib/elixir/lib/macro.ex

```diff
@@ -2522,7 +2523,17 @@ defmodule Macro do
   end
 
   defp dbg_format_ast_to_debug({:value, code_ast, value}, options) do
-    {[dbg_format_ast(code_ast), " ", inspect(value, options), ?\n], value}
+    code = to_string_with_colors(code_ast, options)
+    {[dbg_format_ast(code), " ", inspect(value, options), ?\n], value}
+  end
+
+  defp to_string_with_colors(ast, options) do
+    options = Keyword.take(options, [:syntax_colors])
```

Comment:
This assumes :syntax_colors is a literal at compile time. So I am afraid the previous approach is better. It is fine to have to_string_with_colors, just make sure to prefix it with two underscores and add a TODO :)

Reply:
I don't think it is, since it is now called withing `__dbg__` at runtime?

Reply:
Oh, sorry, you are right. Apologies, I was reading from my phone and misread it. Yes, this is _perfect_.

Reply:
Great! I'll continue working on the formatter itself and open a proper PR.